### PR TITLE
Added github action that tests functionality for PRs

### PR DESCRIPTION
**Pull request by Lemeri123**
Changes;
Trigger on PR: The action will now trigger on pull requests (on: pull_request).
Validation: It checks if the required outputs (accountId, privateKey, publicKey) are valid, and fails if they're missing.

cc.
@hendrikebbers

### CODE REVIEW

**Comment by Lemeri123** [comment]
@hendrikebbers please review this and let me know if I have to make any changes

File: .github/workflows/basic.yml [hunk 1]
```diff
@@ -1,10 +1,12 @@
-name: Simple Usage
+name: PR Functionality Check
```
Comment (Ndacyayisenga-droid) [inline .github/workflows/basic.yml:1]: I think we can leave this  to `Simple Usage`

File: .github/workflows/basic.yml [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 
 on:
-  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
 
 jobs:
-  build:
+  test:
```
Comment (Ndacyayisenga-droid) [inline .github/workflows/basic.yml:9]: I think we can leave this to `build` as well

**Comment by Ndacyayisenga-droid** [comment]
otherwise looks good. Thanks @Lemeri123

**Review by hendrikebbers — APPROVED** [review]

**Pull request merged**